src/drive/web/modules/actions/utils.js

```diff
@@ -0,0 +1,159 @@
+import { forceFileDownload } from 'cozy-stack-client/dist/utils'
+import {
+  saveFileWithCordova,
+  saveAndOpenWithCordova
+} from 'drive/mobile/lib/filesystem'
+
+const { file: fileModel } = models
+const { isDirectory } = fileModel
+
+export const isAnyFileReferencedByAlbum = files => {
+  for (let i = 0, l = files.length; i < l; ++i) {
+    if (isReferencedByAlbum(files[i])) return true
+  }
+  return false
+}
+
+const isMissingFileError = error => error.status === 404
+
+const downloadFileError = error => {
+  return isMissingFileError(error)
+    ? 'error.download_file.missing'
+    : 'error.download_file.offline'
+}
+
+const openFileDownloadError = error => {
+  return isMissingFileError(error)
+    ? 'mobile.error.open_with.missing'
+    : 'mobile.error.open_with.offline'
+}
+
+export const downloadFiles = async (files, client) => {
+  if (files.length === 1 && !isDirectory(files[0])) {
+    const file = files[0]
+
+    try {
+      const downloadURL = await client
+        .collection('io.cozy.files')
+        .getDownloadLinkById(file.id)
+      const filename = file.name
+
+      forceFileDownload(
+        `${client.getStackClient().uri}${downloadURL}?Dl=1`,
+        filename
+      )
+    } catch (error) {
+      Alerter.error(downloadFileError(error))
+      throw error
+    }
+  } else {
+    const ids = files.map(f => f.id)
+    const href = await client
+      .collection('io.cozy.files')
+      .getArchiveLinkByIds(ids)
+    const fullpath = `${client.getStackClient().uri}${href}`
+    forceFileDownload(fullpath, 'files.zip')
+  }
+}
+
+const isAlreadyInTrash = err => {
+  const reasons = err.reason !== undefined ? err.reason.errors : undefined
+  if (reasons) {
+    for (const reason of reasons) {
+      if (reason.detail === 'File or directory is already in the trash') {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+export const trashFiles = async (files, client) => {
+  try {
+    for (const file of files) {
+      await client.collection('io.cozy.files').destroy(file)
+      client.collection('io.cozy.permissions').revokeSharingLink(file)
+    }
+
+    Alerter.success('alert.trash_file_success')
+  } catch (err) {
+    if (!isAlreadyInTrash(err)) {
+      Alerter.error('alert.try_again')
+    }
+  }
+}
+
+export const exportFilesNative = async (files, client, filename) => {
+  const downloadAllFiles = files.map(async file => {
+    const response = await client
+      .collection('io.cozy.files')
+      .fetchFileContent(file)
+
+    const blob = await response.blob()
+    const filenameToUse = filename ? filename : file.name
+    const localFile = await saveFileWithCordova(blob, filenameToUse)
+    return localFile.nativeURL
+  })
+
+  try {
+    Alerter.info('alert.preparing', {
+      duration: Math.min(downloadAllFiles.length * 2000, 6000)
+    })
+    const urls = await Promise.all(downloadAllFiles)
```

Comment:
Nit: we hould use a Promise pool here otherwise we may have problems (too many connections for example) on a case where there are a lot of files.